overlord/hookstate/hookmgr.go

```diff
@@ -283,43 +284,62 @@ func (m *HookManager) undoRunHook(task *state.Task, tomb *tomb.Tomb) error {
 }
 
 func (m *HookManager) runHook(task *state.Task, tomb *tomb.Tomb, snapst *snapstate.SnapState, hooksup *HookSetup) error {
+	_, err := m.runHookReturnContext(task, tomb, snapst, hooksup, nil)
+	return err
+}
+
+func (m *HookManager) EphemeralRunHook(ctx context.Context, hooksup *HookSetup, contextData map[string]interface{}) (*Context, error) {
+	tomb, _ := tomb.WithContext(ctx)
+
+	var snapst snapstate.SnapState
+	m.state.Lock()
+	err := snapstate.Get(m.state, hooksup.Snap, &snapst)
+	m.state.Unlock()
+	if err != nil {
+		return nil, fmt.Errorf("cannot run ephemeral hook %s: %v", hooksup.Hook, err)
+	}
+
+	return m.runHookReturnContext(nil, tomb, &snapst, hooksup, contextData)
+}
+
+func (m *HookManager) runHookReturnContext(task *state.Task, tomb *tomb.Tomb, snapst *snapstate.SnapState, hooksup *HookSetup, contextData map[string]interface{}) (*Context, error) {
```

Comment:
Rename this function, maybe pass in a "loggerFunc" instead of task (because task is only used for the logging).